File: src/command.rs
#[derive(Debug, PartialEq)]
pub enum Command {
    Quit,
    MoveForward,
    MoveBackwards,
    RotateClockwise,
    RotateCounterclockwise,
}

impl From<&str> for Command {
    fn from(value: &str) -> Self {
        match value {
            "0" => Command::Quit,
            "1" => Command::MoveForward,
            "2" => Command::MoveBackwards,
            "3" => Command::RotateClockwise,
            "4" => Command::RotateCounterclockwise,
            other => panic!("Unknown command {}", other),
        }
    }
}

File: src/command.ts
export type Command = "Quit" | "MoveForward" | "MoveBackward" | "RotateClockwise" | "RotateCounterClockwise";

export function commandFromString(value: string): Command {
    switch(value) {
        case "0":
            return "Quit";
        case "1":
            return "MoveForward";
        case "2":
            return "MoveBackward";
        case "3":
            return "RotateClockwise";
        case "4":
            return "RotateCounterClockwise";
        default:
            throw new Error(`Unknown Command '${value}'`);
    }
}
